src/todt.c

```diff
@@ -896,7 +896,7 @@ dt_t **toDtElem(TypeSArray *tsa, dt_t **pdt, Expression *e)
         pdt = dtend(pdt);
         Type *tnext = tsa->next;
         Type *tbn = tnext->toBasetype();
-        while (tbn->ty == Tsarray && (!e || tbn != e->type->nextOf()))
+        while (tbn->ty == Tsarray && (!e || !tbn->equivalent(e->type->nextOf())))
```

Comment:
I'm backporting regressions, here actually: https://github.com/tramker/dmd/tree/2.068.mart
I'm not saying it's a problem, just FYI.